### PR TITLE
Add sub group information to ResourceGroupStateInfo

### PR DESCRIPTION
**Pull request by rongrong**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/server/ResourceGroupStateInfo.java [hunk 1]
```diff
@@ -94,6 +102,18 @@ public DataSize getMemoryUsage()
     }
 
     @JsonProperty
+    public int getNumSubGroups()
+    {
+        return numSubGroups;
+    }
+
+    @JsonProperty
+    public List<ResourceGroupId> getEligibleSubGroups()
```
Comment (raghavsethi) [inline presto-main/src/main/java/com/facebook/presto/server/ResourceGroupStateInfo.java]: What is an 'eligible' subGroup?

Comment (rongrong) [inline presto-main/src/main/java/com/facebook/presto/server/ResourceGroupStateInfo.java]: Eligible sub groups are the subgroups that this resource group would consider when scheduling (meaning that they have a can_run status). So when you think about subgroups that are competing for resource, only eligible sub groups should be considered.

File: presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java [hunk 2]
```diff
@@ -177,7 +177,21 @@ public ResourceGroupStateInfo getStateInfo()
                     runningTimeLimit,
                     queuedTimeLimit,
                     getAggregatedRunningQueriesInfo(),
-                    queuedQueries.size() + descendantQueuedQueries);
+                    queuedQueries.size() + descendantQueuedQueries,
+                    subGroups.values().stream()
+                            .map(subGroup -> new ResourceGroupInfo(
+                                    subGroup.getId(),
+                                    new DataSize(softMemoryLimitBytes, BYTE),
```
Comment (raghavsethi) [inline presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java]: succinctBytes

File: presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java [hunk 3]
```diff
@@ -177,7 +177,21 @@ public ResourceGroupStateInfo getStateInfo()
...
+                                    maxRunningQueries,
+                                    runningTimeLimit,
+                                    maxQueuedQueries,
+                                    queuedTimeLimit,
+                                    getState(),
+                                    eligibleSubGroups.size(),
+                                    new DataSize(cachedMemoryUsageBytes, BYTE),
```
Comment (raghavsethi) [inline presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java]: Same

**Pull request closed without merging**